app/services/attempts_api/request_token_validator.rb

```diff
@@ -0,0 +1,82 @@
+# frozen_string_literal: true
+
+module AttemptsApi
+  class RequestTokenValidator
+    include ActiveModel::Model
+
+    attr_reader :bearer, :issuer, :token
+
+    validates :token, :issuer, :bearer, presence: true
+    validates :bearer, comparison: { equal_to: 'Bearer' }
+    validate :config_data_exists
+    validate :service_provider_exists, if: :config_data_exists?
+    validate :valid_request_token?, if: :config_data_exists?
+
+    def initialize(auth_request_header)
+      @bearer, @issuer, @token = auth_request_header&.split(' ', 3)
+    end
+
+    private
+
+    def config_data_exists
+      return if config_data_exists?
+
+      errors.add(
+        :issuer,
+        :not_authorized,
+        message: 'Issuer is not authorized to use Attempts API',
+      )
+    end
+
+    def issuer_is_authorized
+      errors.add(
+        :issuer,
+        :not_authorized,
+        message: 'Issuer is not authorized to use Attempts API',
+      )
+    end
+
+    def service_provider_exists
+      return if service_provider.present?
+
+      errors.add(
+        :service_provider,
+        :not_authorized,
+        message: 'ServiceProvider does not exist',
+      )
+    end
+
+    def valid_request_token?
```

Comment:
Just for my own reference/note-taking, this is based on some of the previous implementation [here](https://github.com/18F/identity-idp/pull/7888)

Reply:
yes, apologies! 

the big difference is we are hashing the token *before* saving it as config, so we don't have it stored in plaintext anywhere.